crates/typst/src/geom/gradient.rs

```diff
@@ -0,0 +1,1089 @@
+/// ```
+///
+/// ### Rocket
+///
+/// The [`rocket`]($gradient.rocket) gradient is a black to red to white
+/// gradient that is perceptually uniform. Rocket does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.rocket))
+/// ```
+///
+/// ### Mako
+///
+/// The [`mako`]($gradient.mako) gradient is a black to teal to yellow
+/// gradient that is perceptually uniform. Mako does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.mako))
+/// ```
+///
+/// ### Vlag
+///
+/// The [`vlag`]($gradient.vlag) gradient is a light blue to white to red
+/// gradient that is perceptually uniform. Vlag does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.vlag))
+/// ```
+///
+/// ### Icefire
+///
+/// The [`icefire`]($gradient.icefire) gradient is a light teal to black to yellow
+/// gradient that is perceptually uniform. Icefire does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.icefire))
+/// ```
+///
+/// ### Flare
+///
+/// The [`flare`]($gradient.flare) gradient is an orange to purple gradient
+/// that is perceptually uniform. Flare does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.flare))
+/// ```
+///
+/// ### Crest
+///
+/// The [`crest`]($gradient.crest) gradient is a blue to white to red gradient
+/// that is perceptually uniform. Crest does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.crest))
+/// ```
+///
+/// ### Regarding presets like "jet" and "parula"
+///
+/// - [Jet](https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/)
+///   is not a good color map, as it is not perceptually uniform. As such,
+///   it is not color blind friendly and should not be used for data visualization,
+///   due to which it is not included in Typst.
+/// - [Parula](https://www.mathworks.com/help/matlab/ref/parula.html)
+///   is a good color map included in MATLAB, but it is not included as
+///   a preset in Typst. This is because it is owned by MathWorks and is not public.
+///
+#[ty(scope)]
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub enum Gradient {
+    Linear(LinearGradient),
+}
+
+impl Debug for Gradient {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
```

Comment:
```suggestion
    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
```
with import `use std::fmt::{self, Debug, Formatter};`